src/pages/create_blog.jsx

```diff
@@ -0,0 +1,158 @@
+import { useState } from "react";
+import { collection, addDoc } from "@firebase/firestore";
+import { db, storage } from "@/firebase/config";
+import { ref, uploadBytes } from "firebase/storage";
+import Button from "@/components/ui/Button";
+import PageIntro from "@/components/PageIntro";
+import { withTranslation } from "next-i18next";
+import { serverSideTranslations } from "next-i18next/serverSideTranslations";
+import { useRouter } from "next/navigation";
+
+export const extractFirstName = (str) => {
+    const firstSpaceIndex = str.indexOf(" ");
+    if (firstSpaceIndex === -1) {
+        return str;
+    }
+    return str.slice(0, firstSpaceIndex).toLowerCase();
+};
+const CreateBlog = ({ t }) => {
+    const [enTitle, SetEnTitle] = useState("");
+    const [arTitle, setArTitle] = useState("");
+    const [enPostText, setEnPostText] = useState("");
+    const [arpostText, setArPostText] = useState("");
+    const [imageUpload, setImageUpload] = useState(null);
+    const postCollection = collection(db, "blogs");
+    const router = useRouter();
+    const uploadImage = () => {
+        if (imageUpload == null) return;
+        const imageRef = ref(
+            storage,
+            `blogImages/${extractFirstName(enTitle)}.jpeg`
+        );
+        uploadBytes(imageRef, imageUpload).then(() => alert("Image uploaded"));
+    };
+
+    const setData = async () => {
+        try {
+            await uploadImage();
+            await addDoc(postCollection, {
+                ar_title: arTitle,
+                en_title: enTitle,
+                ar_article: arpostText,
+                en_article: enPostText,
+            });
+            await router.push({
+                pathname: "/blog",
+                query: { en_title: enTitle },
+            });
+        } catch (error) {
+            console.error(error);
+            router.push("/404");
+        }
+    };
+
+    return (
+        <>
+            <div className='w-full p-4   flex lg:justify-start sm:items-center sm:justify-center'>
+                {" "}
+                <PageIntro
+                    title={t("uploadYourArticle")}
+                    subtitle={t("AddYouArticleToOurBlogs")}
+                />
+            </div>
+
+            <div className='flex flex-wrap '>
+                <div className='w-full md:w-1/2 p-4 bg-light-cyan  '>
+                    <label className='block mb-2 text-sm font-medium text-gray-900 dark:text-white'>
```

Comment:
use label as prop in input , please check input component in ui

Reply:
already edit it ,done